From f64dec5f5eb63e027b502e9872c949a0fec9047b Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:38:14 -0400
Subject: [virt] kvm/nvmx: Report CPU_BASED_VIRTUAL_NMI_PENDING as supported

Message-id: <1406040016-3289-100-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86178
O-Subject: [RHEL7 PATCH v2 099/221] nVMX: Report CPU_BASED_VIRTUAL_NMI_PENDING as supported
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jan Kiszka <jan.kiszka@siemens.com>

If the host supports it, we can and should expose it to the guest as
well, just like we already do with PIN_BASED_VIRTUAL_NMIS.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit a294c9bbd0dd0dea415a0ee5b8ee9c07f65f5496)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index c77c980..ff46c6b 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -2228,7 +2228,8 @@ static __init void nested_vmx_setup_ctls_msrs(void)
   nested_vmx_procbased_ctls_low, nested_vmx_procbased_ctls_high);
  nested_vmx_procbased_ctls_low = 0;
  nested_vmx_procbased_ctls_high &=
-  CPU_BASED_VIRTUAL_INTR_PENDING | CPU_BASED_USE_TSC_OFFSETING |
+  CPU_BASED_VIRTUAL_INTR_PENDING |
+  CPU_BASED_VIRTUAL_NMI_PENDING | CPU_BASED_USE_TSC_OFFSETING |
   CPU_BASED_HLT_EXITING | CPU_BASED_INVLPG_EXITING |
   CPU_BASED_MWAIT_EXITING | CPU_BASED_CR3_LOAD_EXITING |
   CPU_BASED_CR3_STORE_EXITING |
-- 
1.7.1